src/acp/util/setAgentAccess.test.ts

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright 2021 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import { jest, describe, it, expect } from "@jest/globals";
+import { ACL, ACP } from "../constants";
+import {
+  DEFAULT_ACCESS_CONTROL_RESOURCE_URL,
+  DEFAULT_DOMAIN,
+  TEST_URL,
+} from "../mock/constants";
+import { createDatasetFromSubjects } from "../mock/dataset";
+import { mockAccessControlledResource } from "../mock/mockAccessControlledResource";
+import { setAgentAccess } from "./setAgentAccess";
+
+describe("setAgentAccess()", () => {
+  it("returns the default access modes for an empty ACR", async () => {
+    const resource = mockAccessControlledResource();
+
+    expect(resource.internal_acp.acr.graphs).toStrictEqual({ default: {} });
+    expect(
+      (await setAgentAccess(resource, TEST_URL.defaultWebId, { read: true }))
+        .internal_acp.acr.graphs.default
+    ).toStrictEqual({
+      [DEFAULT_ACCESS_CONTROL_RESOURCE_URL]: {
+        predicates: {
+          [ACP.accessControl]: {
+            namedNodes: [TEST_URL.defaultAccessControl],
+          },
+          [ACP.memberAccessControl]: {
+            namedNodes: [TEST_URL.defaultMemberAccessControl],
+          },
```

Comment:
Should `setAgentAccess` implicitly set member access too ? I would have expected you'd explicitly `setAgentMemberAccess`, or `setAgentAccess(resource, agent, {read: true, memberRead: true}`

Reply:
Good point, but that would be an API redesign. I don't think that is what the current API exposes (given there is no specialisation of the function and I'd imagine it would be called to share resources and containers alike). I might be wrong.

Reply:
I think the current universal API has no notion of membership at all. I would suggest we keep the current behaviour for the universal API and not set membership, and add this in the future as part of the API rework.

Reply:
Oh cool, yes absolutely. I’ll refactor that out!

Reply:
Actually makes tests easier to write 😅
Blessing!

Reply:
\o/

Reply:
Done!